client.go

```diff
@@ -381,9 +384,13 @@ func commitContainerToImage(img, containerID, repo, tag, comment, author string)
 		"CMD " + joinAsArray(info.Config.Cmd, true),
 	}
 
+	var reference string
+	if repo != "" {
+		reference = repo + ":" + tag
+	}
```

Comment:
If we really want to validate those arguments, we should use https://godoc.org/github.com/docker/distribution/reference#ParseNamed instead. The code for that could look similar to the following:
``` golang
named, err := reference.ParseNamed(repo + ":" + tag)
// error checks
ref, isNamedTagged := named.(reference.NamedTagged)
if !isNamedTagged { // error out }
// continue as below
```

Otherwise, we can leave the validation to the `client.ContainerCommit(...)` call, just as we did before.

Reply:
As already discussed in #136, the repo and tag variable can be empty strings if the image is committed from a stopped container. In that case only an image id is provided. Therefore the check if the string is empty.
As you mentioned ContainerCommit(...) already checks if the reference is valid. 
So I suggest leaving it as it is.